com.unity.sg2/Editor/GraphDeltaRegistry/PreviewManager/HeadlessPreviewManager.cs

```diff
@@ -181,6 +183,21 @@ public void SetActiveRegistry(Registry registryInstance)
             m_RegistryInstance = registryInstance;
         }
 
+        /// <summary>
+        /// Returns true if the node needs either its shaders or its preview render regenerated
+        /// </summary>
+        /// <param name="nodeName"> ID of the node we are trying to identify as dirty or not </param>
+        /// <returns></returns>
+        public bool IsNodeDirty(string nodeName)
+        {
+            if (m_CachedPreviewData.TryGetValue(nodeName, out var previewData))
+            {
+                return previewData.isRenderOutOfDate || previewData.isShaderOutOfDate;
+            }
+
+            return false;
+        }
```

Comment:
This feels weird as a public action and sort of works against the Async styled interface.

UI should just be requesting things and it will get them whenever it gets them. It doesn't seem like it would need to ask if things are in a dirty state or not.